docs/src/DeveloperDocumentation/styleguide.md

```diff
@@ -241,3 +217,249 @@ existing block.
+
+```
+
+##### Detailed printing in a single line.
+
+This version needs to be used in case the detailed
+printing does not contain newlines.
+Then detailed and one line printing agree.
+The `if` clause takes care of supercompact printing as well.
+
+```julia
+struct NewRing2
+  base_ring
+end
+
+base_ring(R::NewRing2) = R.base_ring
+
+function Base.show(io::IO, R::NewRing2)
+  if get(io, :supercompact, false)
+    # no nested printing
+    print(io, "supercompact printing of newring")
+  else
+    # nested printing allowed, preferably supercompact
+    print(io, "I am a new ring and always print in one line " )
+    print(IOContext(io, :supercompact => true), base_ring(R))
+  end
+end
+```
+
+```julia
+julia> R = NewRing2(QQ)
+I am a new ring and always print in one line QQ
+
+julia> [R,R]
+2-element Vector{NewRing2}:
+ I am a new ring and always print in one line Rational Field
+ I am a new ring and always print in one line Rational Field
+
+julia> print(IOContext(Base.stdout, :supercompact => true) ,R)
+supercompact printing of newring
+```
+
+##### The following is not working as expected and should not be used
+
+```julia
+function Base.show(io::IO, ::MIME"text/plain", R::NewRing)  # do not implement me like this
+  print(io, "I am a new ring with a detailed printing of one line")
+end
+```
+
+Then the following will *not* be used for array/tuple printing.
+It will be used for `print(io, R::NewRing)` though.
+
+```julia
+function Base.show(io::IO, R::NewRing)
+  if get(io, :supercompact, false)
+    print(io, "supercompact printing of newring")
+  else # this is what we call one line
+    print(io, "one line printing of newring with ")
+    print(IOContext(io, :supercompact => true), "supercompact ", R.base_ring)
+  end
+end
+```
+This example illustrates the unexpected behavior.
+```julia
+julia> R = NewRing(1)
+
+julia> R
+I am a new ring with a detailed printing of one line
+
+julia> [R,R]  # one line printing is ignored
+2-element Vector{NewRing}:
+ I am a new ring with a detailed printing of one line
+ I am a new ring with a detailed printing of one line
+
+julia> print(Base.stdout, R)
+one line printing of newring with supercompact QQ
+```
+
+## LaTeX and Unicode printing
+
+### LaTeX output
+Some types support LaTeX output.
+```
+julia> Qx, x = QQ["x"];
+
+julia> show(stdout, "text/latex", x^2 + 2x + x^10)
+x^{10} + x^{2} + 2 x
+
+julia> show(stdout, "text/latex", Qx[x x^2; 1 1])
+\begin{array}{cc}
+x & x^{2} \\
+1 & 1
+\end{array}
+```
+
+```
+Base.show(io::IOContext, ::MIME"text/latex")
+```
+
+### Unicode printing
+Per default output should be ANSI only (no Unicode). Implementors of
+`Base.show` and related functions can branch on the output of
+`Oscar.is_unicode_allowed()` to display objects using non-ASCII characters.
+This will then be used for users which enabled Unicode using
+`allow_unicode(true)`. Note that
+
+- there must be a default ANSI only output, since this is the default setting
```

Comment:
```suggestion
- there must be a default ASCII only output, since this is the default setting
```
There are many ANSI encodings, ASCII is just one of them